.clang-tidy

```diff
@@ -20,18 +20,23 @@ Checks: [
 
   '-bugprone-assignment-in-if-condition',
   '-bugprone-branch-clone',
+  '-bugprone-crtp-constructor-accessibility',
   '-bugprone-easily-swappable-parameters',
   '-bugprone-exception-escape',
   '-bugprone-implicit-widening-of-multiplication-result',
   '-bugprone-multi-level-implicit-pointer-conversion',
   '-bugprone-narrowing-conversions',
-  '-bugprone-unchecked-optional-access',
-  '-bugprone-crtp-constructor-accessibility',
+  '-bugprone-nondeterministic-pointer-iteration-order',
```

Comment:
This one looks like an obscure edge case. Perhaps there is just a handful of instances which we can fix right away?